File: math/reverse-integer/reverse-integer.ts
function reverse(x: number): number {
  const max: number = 2 ** 31 - 1;
  const min: number = -(2 ** 31);

  let rev = 0;

  while (x !== 0) {
    let add = x % 10;

    x = ~~(x / 10);

    if (rev > max / 10 || (rev == max / 10 && add > 7)) {
      return 0;
    }

    if (rev < min / 10 || (rev == min / 10 && add < -8)) {
      return 0;
    }

    rev = rev * 10 + add;
  }

  return rev;
}

File: math/reverse-integer/reverse-integer.go
func reverse(x int) int {
	rev := 0

	for x != 0 {
		add := x % 10

		x /= 10

		if rev > math.MaxInt32/10 || (rev == math.MaxInt32/10 && add > 7) {
			return 0
		}

		if rev < math.MinInt32/10 || (rev == math.MinInt32/10 && add < -8) {
			return 0
		}

		rev = rev*10 + add
	}

	return rev
}
